pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,32 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
+	}
+
+	anonAccessMode := obj.Spec.Parameters["AnonymousAccessMode"]
```

Comment:
This is part of Spec https://github.com/container-object-storage-interface/api/blob/fb97594fb7a4fa6c664c79075a52255e14dd8fba/apis/objectstorage.k8s.io/v1alpha1/types.go#L47